src/gt4py/frontend/gtscript_frontend.py

```diff
@@ -352,17 +352,20 @@ class CallInliner(ast.NodeTransformer):
     """
 
     @classmethod
-    def apply(cls, func_node: ast.FunctionDef, context: dict):
+    def apply(
+        cls, func_node: ast.FunctionDef, context: dict, *, call_stack: Optional[Set[str]] = None
+    ):
         inliner = cls(context)
-        inliner(func_node)
+        inliner(func_node, call_stack=call_stack or set())
         return inliner.all_skip_names
 
     def __init__(self, context: dict):
         self.context = context
         self.current_block = None
         self.all_skip_names = set(gtscript.builtins) | {"gt4py", "gtscript"}
 
-    def __call__(self, func_node: ast.FunctionDef):
+    def __call__(self, func_node: ast.FunctionDef, *, call_stack: Set[str]):
+        self.call_stack = call_stack
```

Comment:
Ah, right. The problem is that `ast.NodeTransformer` does not pass `**kwargs` through all methods, since its `generic_visit` function does not have `**kwargs` in the signature.